components/dashboard/src/settings/EnvironmentVariables.tsx

```diff
@@ -4,12 +4,195 @@
+    };
+
+    return <SettingsPage title='Variables' subtitle='Configure environment variables for all workspaces.'>
+        {isAddEnvVarModalVisible ? <AddEnvVarModal
+            save={save}
+            envVar={currentEnvVar}
+            validate={validate}
+            onClose={() => setAddEnvVarModalVisible(false)} /> : null}
+        {envVars.length === 0
+            ? <div className="bg-gray-100 rounded-xl w-full h-96">
+                <div className="pt-28 flex flex-col items-center w-96 m-auto">
+                    <h3 className="text-center pb-3">No Environment Variables</h3>
+                    <div className="text-center pb-6 text-gray-500">In addition to user-specific environment variables you can also pass variables through a workspace creation URL. <a className="text-gray-400 underline underline-thickness-thin underline-offset-small hover:text-gray-600" href="https://www.gitpod.io/docs/env-vars/">Learn more</a></div>
+                    <button onClick={add} className="font-medium">New Environment Variable</button>
+                </div>
+            </div>
+            : <div>
+                <div className="flex justify-end mb-2">
+                    <button onClick={add} className="ml-2 font-medium">New Environment Variable</button>
+                </div>
+                <div className="flex flex-col space-y-2">
+                    <div className="px-3 py-3 flex justify-between space-x-2 text-sm text-gray-400 border-t border-b border-gray-200">
+                        <div className="w-5/12">Name</div>
+                        <div className="w-5/12">Scope</div>
+                        <div className="w-2/12"></div>
+                    </div>
+                </div>
+                <div className="flex flex-col">
+                    {envVars.map(ev => {
+                        return <div className="rounded-xl whitespace-nowrap flex space-x-2 py-3 px-3 w-full justify-between hover:bg-gray-100 focus:bg-gitpod-kumquat-light group">
+                            <div className="w-5/12 m-auto">{ev.name}</div>
+                            <div className="w-5/12 m-auto text-sm text-gray-400">{ev.repositoryPattern}</div>
+                            <div className="w-2/12 flex justify-end">
+                                <div className="flex w-8 self-center hover:bg-gray-200 rounded-md cursor-pointer">
+                                    <ContextMenu menuEntries={[
+                                        {
+                                            title: 'Edit',
+                                            onClick: () => edit(ev),
+                                            separator: true
+                                        },
+                                        {
+                                            title: 'Delete',
+                                            customFontStyle: 'text-red-600 hover:text-red-800',
+                                            onClick: () => deleteV(ev)
```

Comment:
**issue**: We should definitely add a confirmation modal here! What do you think?

> **Title**: Delete Variable
> **Body**: Are you sure you want to delete this environment variable?
> **Actions**: Cancel (Secondary), Delete. Variable (Primary)

| Draft Prototype |
|-|
| <img width="498" alt="Screenshot 2021-03-25 at 6 13 48 PM" src="https://user-images.githubusercontent.com/120486/112505915-ea91cf80-8d95-11eb-8815-1e764968f870.png"> |

Reply:
filed https://github.com/gitpod-io/gitpod/issues/3604